defaults.go

```diff
@@ -3,6 +3,7 @@ package sprig
 import (
 	"bytes"
 	"encoding/json"
+	"gopkg.in/yaml.v3"
```

Comment:
you'd better use "sigs.k8s.io/yaml" to supoort "omitempty" json tags

Reply:
Could you elaborate why we would benefit from `omitempty` in this context? 
We can't specify tags during the templating

Reply:
AFAICT [the Helm version of this](https://github.com/helm/helm/blob/main/pkg/engine/funcs.go#L85) doesn't specifically use `omitempty` either, so not sure whether that is really relevant. It _does_, however, make sense to use the same parser that Helm does, in case there are other differences in implementation.

(note that Helm uses a different [yaml engine](https://github.com/helm/helm/blob/main/pkg/engine/funcs.go#L28) for `toYamlPretty`